src/components/Send.jsx

```diff
@@ -1,17 +1,104 @@
+      <rb.Form.Label className="mb-0">Choose a number of collaborators ({numCollaborators})</rb.Form.Label>
+      <div className="mb-2">
+        <rb.Form.Text className="text-secondary">
+          A higher number is better for privacy, but also increases the fee.
+        </rb.Form.Text>
+      </div>
+      <div className="d-flex flex-row flex-wrap">
+        {collaboratorsSelection.map((number, index) => {
+          return (
+            <rb.Button
+              key={index}
+              variant="white"
+              className={`py-2 px-2 border rounded text-center${
+                !usesCustomNumCollaborators && numCollaborators === number ? ' border-dark border-2' : ''
+              }`}
+              onClick={() => {
+                setUsesCustomNumCollaborators(false)
+                setNumCollaborators(number)
+              }}
+            >
+              {number}
+            </rb.Button>
+          )
+        })}
+        <rb.Form.Control
+          type="number"
+          min={1}
+          placeholder="Other"
+          className={`py-2 px-2 border rounded text-center${usesCustomNumCollaborators ? ' border-dark border-2' : ''}`}
+          onChange={(e) => {
+            setUsesCustomNumCollaborators(true)
+            setNumCollaborators(e.target.value)
+          }}
+          onClick={(e) => {
+            if (e.target.value) {
+              setUsesCustomNumCollaborators(true)
+              setNumCollaborators(e.target.value)
+            }
+          }}
+        />
+      </div>
+      <rb.Form.Control.Feedback type="invalid">Please set the counterparties.</rb.Form.Control.Feedback>
+    </rb.Form.Group>
+  )
+}
+
+export default function Send({ currentWallet }) {
+  const wallet = useCurrentWallet()
+  const walletInfo = useCurrentWalletInfo()
+  const setWalletInfo = useSetCurrentWalletInfo()
+  const settings = useSettings()
+
   const location = useLocation()
   const [validated, setValidated] = useState(false)
   const [alert, setAlert] = useState(null)
   const [isSending, setIsSending] = useState(false)
   const [isCoinjoin, setIsCoinjoin] = useState(false)
   const [account, setAccount] = useState(parseInt(location.state?.account, 10) || 0)
+  const [numCollaborators, setNumCollaborators] = useState(6) // Todo: Sane default
+
+  useEffect(() => {
+    // Reload wallet info if not already available.
+    if (walletInfo) return
+
+    const abortCtrl = new AbortController()
+    const { name, token } = wallet
+    const opts = {
+      headers: { Authorization: `Bearer ${token}` },
+      signal: abortCtrl.signal,
+    }
+
+    setAlert(null)
+
+    fetch(`/api/v1/wallet/${name}/display`, opts)
+      .then((res) => (res.ok ? res.json() : Promise.reject(new Error(res.message || 'Loading wallet failed.'))))
+      .then((data) => setWalletInfo(data.walletinfo))
+      .catch((err) => {
+        !abortCtrl.signal.aborted && setAlert({ variant: 'danger', message: err.message })
+      })
```

Comment:
Can now also use `Api.getWalletDisplay` if you want to.

Reply:
Nice! 🙌 Will do.